kafka/consumerHandler.go

```diff
@@ -0,0 +1,70 @@
+// Package kafka provides producer and consumer to work with kafka topics
+package kafka
+
+import (
+	"encoding/json"
+	"log"
+
+	"github.com/Shopify/sarama"
+	"github.com/lvl484/positioning-filter/matcher"
+	"github.com/lvl484/positioning-filter/position"
+)
+
+type consumerGroupHandler struct {
+	controller messageController
+}
+
+// Setup is used to make ConsumerGroupHandler implement sarama.ConsumerGroupHandler interface
+func (consumerGroupHandler) Setup(_ sarama.ConsumerGroupSession) error { return nil }
+
+// Setup is used to make ConsumerGroupHandler implement sarama.ConsumerGroupHandler interface
+func (consumerGroupHandler) Cleanup(_ sarama.ConsumerGroupSession) error { return nil }
+
+// ConsumeClaim consume messages from kafka topic within consumer group
+func (h consumerGroupHandler) ConsumeClaim(sess sarama.ConsumerGroupSession, claim sarama.ConsumerGroupClaim) error {
+	for msg := range claim.Messages() {
+		var p position.Position
+		if err := json.Unmarshal(msg.Value, &p); err != nil {
+			log.Println()
```

Comment:
In case when message is invalid you should mark message

Reply:
fixed